Attributes following a missing trailing comma produce over-eager recovery

https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=8a65ac81b182494a4856b59164bffbc2

```rust
struct Feelings {
    owo: bool
    #[allow(unused)] // If commented-out, we recover differently and do not get the additional error about no field `owo`
    uwu: bool,
}

impl Feelings {
    fn hmm(&self) -> bool {
        self.owo
    }
}
```

Having an attribute after the field with missing `,` turns the error from this (which is good):
```
   Compiling playground v0.0.1 (/playground)
error: expected `,`, or `}`, found `uwu`
 --> src/main.rs:2:14
  |
2 |     owo: bool
  |              ^ help: try adding a comma: `,`

error: could not compile `playground` due to previous error
```
To this, which contains unhelpful errors and possibly a huge number of them on a larger codebase:
```
Compiling playground v0.0.1 (/playground)
error: expected `,`, or `}`, found `#`
 --> src/main.rs:2:14
  |
2 |     owo: bool
  |              ^

error[E0609]: no field `owo` on type `&Feelings`
 --> src/main.rs:9:14
  |
9 |         self.owo
  |              ^^^ unknown field

For more information about this error, try `rustc --explain E0609`.
error: could not compile `playground` due to 2 previous errors
```

<!-- TRIAGEBOT_START -->

<!-- TRIAGEBOT_ASSIGN_START -->

<!-- TRIAGEBOT_ASSIGN_DATA_START$${"user":"chenyukang"}$$TRIAGEBOT_ASSIGN_DATA_END -->

<!-- TRIAGEBOT_ASSIGN_END -->
<!-- TRIAGEBOT_END -->

--- a/compiler/rustc_parse/src/parser/item.rs
+++ b/compiler/rustc_parse/src/parser/item.rs
@@ -1539,8 +1539,12 @@ impl<'a> Parser<'a> {
                     }
                 }
 
-                if self.token.is_ident() {
-                    // This is likely another field; emit the diagnostic and keep going
+                if self.token.is_ident()
+                    || (self.token.kind == TokenKind::Pound
+                        && (self.look_ahead(1, |t| t == &token::OpenDelim(Delimiter::Bracket))))
+                {
+                    // This is likely another field, TokenKind::Pound is used for `#[..]` attribute for next field,
+                    // emit the diagnostic and keep going
                     err.span_suggestion(
                         sp,
                         "try adding a comma",

--- a/src/test/ui/parser/struct-filed-with-attr.fixed
+++ b/src/test/ui/parser/struct-filed-with-attr.fixed
@@ -1,0 +1,18 @@
+// Issue: 100461, Try to give a helpful diagnostic even when the next struct field has an attribute.
+// run-rustfix
+
+struct Feelings {
+    owo: bool,
+    //~^ ERROR expected `,`, or `}`, found `#`
+    #[allow(unused)]
+    uwu: bool,
+}
+
+impl Feelings {
+    #[allow(unused)]
+    fn hmm(&self) -> bool {
+        self.owo
+    }
+}
+
+fn main() { }

--- a/src/test/ui/parser/struct-filed-with-attr.rs
+++ b/src/test/ui/parser/struct-filed-with-attr.rs
@@ -1,0 +1,18 @@
+// Issue: 100461, Try to give a helpful diagnostic even when the next struct field has an attribute.
+// run-rustfix
+
+struct Feelings {
+    owo: bool
+    //~^ ERROR expected `,`, or `}`, found `#`
+    #[allow(unused)]
+    uwu: bool,
+}
+
+impl Feelings {
+    #[allow(unused)]
+    fn hmm(&self) -> bool {
+        self.owo
+    }
+}
+
+fn main() { }

--- a/src/test/ui/parser/struct-filed-with-attr.stderr
+++ b/src/test/ui/parser/struct-filed-with-attr.stderr
@@ -1,0 +1,8 @@
+error: expected `,`, or `}`, found `#`
+  --> $DIR/struct-filed-with-attr.rs:5:14
+   |
+LL |     owo: bool
+   |              ^ help: try adding a comma: `,`
+
+error: aborting due to previous error
+

